x/evm/keeper/statedb.go

```diff
@@ -571,16 +571,18 @@ func (k *Keeper) AddSlotToAccessList(addr common.Address, slot common.Hash) {
 // Snapshotting
 // ----------------------------------------------------------------------------
 
-// Snapshot return zero as the state changes won't be committed if the state transition fails. So there
-// is no need to snapshot before the VM execution.
-// See Cosmos SDK docs for more info: https://docs.cosmos.network/master/core/baseapp.html#delivertx-state-updates
+// Snapshot return the index in the cached context stack
```

Comment:
this should describe that it also appends a new cache context to the stack

Reply:
done